common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java

```diff
@@ -366,59 +448,105 @@ private void validateEncodeVersion(String encryptedBlob, int encodeVersionLength
     @Override
     public synchronized SecretKey loadSecretKeyForEncryption() throws IOException,
             GeneralSecurityException {
-        final String pkgName = mContext.getPackageName();
-        final byte[] secretKeyData = getSecretKeyData(pkgName);
-        return loadSecretKeyForEncryption(secretKeyData == null ? VERSION_ANDROID_KEY_STORE : VERSION_USER_DEFINED);
+        // Loading key only once for performance. If API is upgraded, it will
+        // restart the device anyway. It will load the correct key for new API.
+        if (mEncryptionKey != null && mEncryptionHMACKey != null) {
+            return mEncryptionKey;
+        }
+
+        final String packageName = getPackageName();
+        if (COMPANY_PORTAL_APP_PACKAGE_NAME.equalsIgnoreCase(packageName) ||
```

Comment:
Does this mean the Authenticator and Company Portal will be using these keys too for their local auth?

Reply:
Good catch. I'll simplify the logic to do just the following 

1. look for ADAL secret Key -> if exist, returns.
2. Use KeyStore secret key.

No need to separate cases anymore as we're no longer re-encrypt inline.